sdk/keyvault/keyvault-certificates/test/CRUD.test.ts

```diff
@@ -272,37 +272,68 @@ describe("Certificates client - create, read, update and delete", () => {
     );
   });
 
-  it("can get a deleted certificate", async function() {
-    const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
-    await client.beginCreateCertificate(
-      certificateName,
-      basicCertificatePolicy,
-      testPollerProperties
-    );
-    const deletePoller = await client.beginDeleteCertificate(certificateName, testPollerProperties);
-    const deletedCertificate = await deletePoller.pollUntilDone();
-    assert.equal(
-      deletedCertificate.properties.name,
-      certificateName,
-      "Unexpected certificate name in result from getCertificate()."
-    );
-    await testClient.purgeCertificate(certificateName);
-  });
+  describe("can get a deleted certificate", () => {
+    it("using beginDeleteCertificate's poller", async function() {
+      const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
+      await client.beginCreateCertificate(
+        certificateName,
+        basicCertificatePolicy,
+        testPollerProperties
+      );
 
-  it("can get a deleted certificate (Non Existing)", async function() {
-    const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
-    let error;
-    try {
-      await client.beginDeleteCertificate(certificateName, testPollerProperties);
-      throw Error("Expecting an error but not catching one.");
-    } catch (e) {
-      error = e;
-    }
-    assert.equal(
-      error.message,
-      `Certificate not found: ${certificateName}`,
-      "Unexpected certificate name in result from getKey()."
-    );
+      const deletePoller = await client.beginDeleteCertificate(
+        certificateName,
+        testPollerProperties
+      );
+      let deletedCertificate = await deletePoller.pollUntilDone();
```

Comment:
Is this test supposed to make sure that you can get the name of a certificate that's already finished being deleted?

Reply:
This test is to check that we effectively get the certificate we're expecting to receive after it gets deleted. I'm only picking the name because I'm sure it's unique every time.

Reply:
In that case, do you need to call deletePoller.pollUntilDone() against once the current one finishes to ensure it's getting the name from an already deleted certificate?

Reply:
I don't think I understood the question. I can tell you this: If we don't wait until the certificate is deleted, we can't read it as a deleted certificate.

Reply:
Sorry, what I meant was, the poller is created before the certificate is deleted and then you wait for it to be deleted. I don't know how the poller works internally, but if the goal is to ensure that you can get the name of an already deleted certificate from the poller returned by `beginDeleteCertificate`, then wouldn't you want to call `beginDeleteCertificate` again after the certificate is confirmed to be deleted?

That change would make it behave similarly to the test below where you call `getDeletedCertificate` after you've finished deleting it.

Reply:
Calling `beginDeleteCertificate` on a deleted certificate will fail 🌞 Let's review how the poller works together. I'd love your feedback there too!